java-checks-common/src/main/java/org/sonar/java/checks/helpers/ReassignmentFinder.java

```diff
@@ -102,11 +102,8 @@ private static Optional<AssignmentExpressionTree> checkAssignment(IdentifierTree
       nonParenthesisParent = previousTree.parent();
     }
 
-    if (nonParenthesisParent instanceof AssignmentExpressionTree) {
-      AssignmentExpressionTree assignment = (AssignmentExpressionTree) nonParenthesisParent;
-      if (assignment.variable().equals(previousTree)) {
-        return Optional.of(assignment);
-      }
+    if (nonParenthesisParent instanceof AssignmentExpressionTree assignment && (assignment.variable().equals(previousTree))) {
```

Comment:
I believe you don't need extra parenthesis around `(assignment.variable().equals(previousTree))`